template/src/components/pressables/BaseButton.tsx

```diff
@@ -0,0 +1,112 @@
+import React, { ReactElement } from 'react'
+import {
+  ActivityIndicator,
+  Insets,
+  Pressable,
+  StyleProp,
+  StyleSheet,
+  Text,
+  TextStyle,
+  View,
+  ViewStyle,
+} from 'react-native'
+import colors from '@config/ui/colors'
+import { FC } from '@utils/types'
+
+const BORDER_RADIUS = 5
+
+type RoundButtonType = 'primary' | 'secondary'
+
+export type BaseButtonPropsType = {
+  isDisabled?: boolean | null
+  onPress?: () => void
+  iconAtStart?: ReactElement
+  iconAtEnd?: ReactElement
+  text?: string | ReactElement
```

Comment:
I see you stopped at the 5th iteration of the Button. Take a look at the problems described further: https://www.smashingmagazine.com/2023/01/key-good-component-design-selfishness/#iteration-6 and the final form: hthttps://www.smashingmagazine.com/2023/01/key-good-component-design-selfishness/#me-myself-and-ui

Reply:
I think now done in favour of flexibility